### PR TITLE
Update mkdirp to 0.5.3 which removes the vulnerable dep on minimist

### PR DESCRIPTION
**Pull request by mjziolko**
https://npmjs.com/advisories/1179

mkdirp has a dependency on minimist which has a prototype pollution vulnerability.  Manually updating to 0.5.3 fixes this issue.

### CODE REVIEW

**Comment by chris-williams-efr** [comment]
commenting to watch progress

**Comment by jrichardsz** [comment]
- current status : failed
- https://ci.appveyor.com/project/Mapbox/node-pre-gyp

I tried to get the error details with no luck. How can I help you?

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
   "main": "./lib/node-pre-gyp.js",
   "dependencies": {
     "detect-libc": "^1.0.2",
-    "mkdirp": "^0.5.1",
```
Comment (Naktibalda) [inline package.json:24]: This change is unnecessary `^0.5.1` allows any version lower than 0.6.

In fact, I checked out node-pre-gyp repo, ran `npm install` and got a clean `npm audit report`:

```
node-pre-gyp$ npm audit --registry https://registry.npmjs.org/

                       === npm audit security report ===

found 0 vulnerabilities
 in 338 scanned packages

node-pre-gyp$ npm ls minimist
node-pre-gyp@0.14.0 /c/Users/Eckoh/sites/node-pre-gyp
├─┬ mkdirp@0.5.4
│ └── minimist@1.2.5
├─┬ rc@1.2.8
│ └── minimist@1.2.5  deduped
└─┬ tape@4.13.2
  └── minimist@1.2.5  deduped

npm ls mkdirp
node-pre-gyp@0.14.0 /c/Users/Eckoh/sites/node-pre-gyp
├── mkdirp@0.5.4
├─┬ nock@9.6.1
│ └── mkdirp@0.5.4  deduped
└─┬ tar@4.4.13
  └── mkdirp@0.5.4  deduped


```

**Comment by Naktibalda** [comment]
This change is unnecessary `^0.5.1` allows any version lower than 0.6.

In fact, I checked out node-pre-gyp repo, ran `npm install` and got a clean `npm audit report`:

```
node-pre-gyp$ npm audit --registry https://registry.npmjs.org/

                       === npm audit security report ===

found 0 vulnerabilities
 in 338 scanned packages

node-pre-gyp$ npm ls minimist
node-pre-gyp@0.14.0 /c/Users/Eckoh/sites/node-pre-gyp
├─┬ mkdirp@0.5.4
│ └── minimist@1.2.5
├─┬ rc@1.2.8
│ └── minimist@1.2.5  deduped
└─┬ tape@4.13.2
  └── minimist@1.2.5  deduped

npm ls mkdirp
node-pre-gyp@0.14.0 /c/Users/Eckoh/sites/node-pre-gyp
├── mkdirp@0.5.4
├─┬ nock@9.6.1
│ └── mkdirp@0.5.4  deduped
└─┬ tar@4.4.13
  └── mkdirp@0.5.4  deduped


```

**Comment by karlhorky** [comment]
> This change is unnecessary `^0.5.1` allows any version lower than 0.6.

This change removes the possibility for `0.5.1` or `0.5.2` to be selected as a transitive dependency, forcing npm and Yarn to resolve to at least `0.5.3`.

These previous versions may be saved in lockfiles. Consider that many users do not know that they can upgrade transitive dependencies. So for those users who have one of the older versions saved in a lockfile, this change will bump this version for them. If one of their top-level dependencies depends on this change, then they just need to upgrade the top-level dependency.

So I would vote for making this change.

**Comment by jrichardsz** [comment]
Error continues. Github says:  

Upgrade minimist to version 0.2.1 or later.

And bcrypt continues using:  0.0.8
├─┬ bcrypt@4.0.1
│ └─┬ node-pre-gyp@0.14.0
│   ├─┬ mkdirp@0.5.1
│   │ └── minimist@0.0.8 
│   └─┬ rc@1.2.8
│     └── minimist@1.2.0

**Comment by silkfire** [comment]
@jrichardsz That issue is due to the `node-pre-gyp` dependency using an outdated version of `mkdirp`. See the tracked issue here: https://github.com/mapbox/node-pre-gyp/issues/495

**Pull request merged**